ci-operator/step-registry/hypershift/mce/power/create/ibmcloud-workers/infra/hypershift-mce-power-create-ibmcloud-workers-infra-commands.sh

```diff
@@ -0,0 +1,233 @@
+  fi
+  if [[ -z "${VPC_ZONE}" ]]; then
+        VPC_ZONE=$(jq -r '.vpcZone' "${AGENT_POWER_CREDENTIALS}/ibmcloud-resources-het.json")
+  fi
+  if [[ -z "${VPC_NAME}" ]]; then
+        VPC_NAME=$(jq -r '.vpcName' "${AGENT_POWER_CREDENTIALS}/ibmcloud-resources-het.json")
+  fi
+  if [[ -z "${VPC_SUBNET_ID}" ]]; then
+        VPC_SUBNET_ID=$(jq -r '.vpcSubnetID' "${AGENT_POWER_CREDENTIALS}/ibmcloud-resources-het.json")
+  fi
+
+}
+
+create_x86_vms() {
+  # IBM cloud login
+  echo | ibmcloud login --apikey @"${AGENT_POWER_CREDENTIALS}/ibmcloud-apikey" -r "${VPC_REGION}"
+
+  # Target resource group
+  ibmcloud target -g "${RESOURCE_GROUP_NAME}"
+
+  # Install vpc plugin
+  ibmcloud plugin install vpc-infrastructure
+
+  # Creating x86 VM in VPC
+  echo "$(date) Creating x86 VSI for VPC"
+  INSTANCE_NAMES=()
+  if [ ${HYPERSHIFT_NODE_COUNT} -eq 1 ]; then
+    INSTANCE_NAMES+=("${VPC_VSI_NAME}")
+  else
+    for (( i = 1; i <= ${HYPERSHIFT_NODE_COUNT}; i++ )); do
+      INSTANCE_NAMES+=("${VPC_VSI_NAME}-${i}")
+    done
+  fi
+  IP_X86_ADDRESSES=()
+  for instance in "${INSTANCE_NAMES[@]}"; do
+    ibmcloud is instance-create $instance "${VPC_NAME}" "${VPC_ZONE}" ${PROFILE_NAME} "${VPC_SUBNET_ID}" --image "${IMAGE_ID}" --keys "${SSH_KEY_ID}"
+    sleep 5
+    instance_ip=$(ibmcloud is instance $instance --output JSON | jq -r '.primary_network_attachment.primary_ip.address')
+    IP_X86_ADDRESSES+=("$instance_ip")
+  done
+
+  # Save VMs information to ${SHARED_DIR} for use in other scenarios.
+  echo "${IP_X86_ADDRESSES[@]}" > "${SHARED_DIR}/ipx86addr"
+  echo "${INSTANCE_NAMES[@]}" > "${SHARED_DIR}/ipx86names"
+  
+}
+
+update_agentserviceconfig() {
+  # Updating AgentServiceConfig with x86 osImages
+  CLUSTER_VERSION=$(oc get clusterversion -o jsonpath={..desired.version} | cut -d '.' -f 1,2)
+  OS_IMAGES=$(jq --arg CLUSTER_VERSION "${CLUSTER_VERSION}" '[.[] | select(.openshift_version == $CLUSTER_VERSION)]' "${SHARED_DIR}/default_os_images.json")
+  # shellcheck disable=SC2034
+  VERSION=$(echo "$OS_IMAGES" | jq -r '.[] | select(.cpu_architecture == "x86_64").version')
+  # shellcheck disable=SC2034
+  URL=$(echo "$OS_IMAGES" | jq -r '.[] | select(.cpu_architecture == "x86_64").url')
+  echo "$(date) Updating AgentServiceConfig"
+  oc patch AgentServiceConfig agent --type=json -p="[{\"op\": \"add\", \"path\": \"/spec/osImages/-\", \"value\": {\"openshiftVersion\": \"${CLUSTER_VERSION}\", \"version\": \"${VERSION}\", \"url\": \"${URL}\",  \"cpuArchitecture\": \"x86_64\"}}]"
+  oc get AgentServiceConfig agent -o yaml
+
+  oc wait --timeout=10m --for=condition=DeploymentsHealthy agentserviceconfig agent
+  echo "$(date) AgentServiceConfig updated"
+
+}
+
+
+set_power_configs() {
+  # PowerVS VSI(Virtual Server Instance) configs
+  POWERVS_VSI_NAME="power-${HOSTED_CLUSTER_NAME}-worker"
+  POWERVS_VSI_MEMORY=16
+  POWERVS_VSI_PROCESSORS=0.5
+  POWERVS_VSI_PROC_TYPE="shared"
+  POWERVS_VSI_SYS_TYPE="e980"
```

Comment:
Let's use s922 only here for power flow.

Reply:
Done.